### ISSUE TITLE
AML: Handle DefIndexField (real hardware)

### ISSUE THREAD

**Issue by alnyan** [rust-osdev/acpi#184]
Hi, 
I was trying to integrate the `aml`/`acpi` crates into my kernel and found out that some of the real-hardware opcodes are NYI. 

I'm currently working on implementing `DefIndexField` myself, but there's problem that might require significant changes to the `aml` crate: as `IndexField`s require writes to their index fields even when reading, this means functions like `as_integer/as_string/as_buffer`/etc as well as the `read_field` also perform writes and thus require mutable ref to `AmlContext`, which breaks lots of other code. A simple workaround would be to add a separate `AmlType` for `IndexField`s, but I don't think this is correct as per specification.

So, if you can suggest which way is better, I can finish this feature and make a PR.
Thanks.

**Comment by IsaacWoods** [comment]
Yes it is - thanks. This was actually one of the problems that forced a fundamental redesign of the interpreter and object representation (albeit with poor communication, apologies to alnyan) so these issues do not apply to the new interpreter design.
